src/components/web3/ExceptionListPanel.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ReactNode, useState, Dispatch } from "react";
+import { bodyText } from "./styles";
+
+interface Props {
+  header: string;
+  subhead: string;
+  web3Address: string;
+  web3Account: "ETH" | "SOL" | null;
+  isExceptionAddressWrong?: boolean;
+  setIsExceptionAddressWrong?: (val: boolean) => void;
+  children?: ReactNode;
+  exceptionList: string[];
+  compareList: string[];
+  onChange: Dispatch<string[]>;
+}
+
+export const ExceptionListPanel: React.FC<Props> = ({
+  header,
+  subhead,
+  web3Account,
+  web3Address,
+  isExceptionAddressWrong,
+  setIsExceptionAddressWrong,
+  exceptionList,
+  compareList,
+  onChange,
+}) => {
+  const [inputText, setInputText] = useState("");
+  const [invalidAddresses, setInvalidAddresses] = useState<string[]>([]);
+  const [validAddresses, setValidAddresses] = useState<string[]>([]);
+
+  const placeholderText =
+    web3Account === "ETH"
+      ? "0x8d247168113b8aa0f020b2441e98b613b9878qmc, ..."
+      : "BtsXBPrf69qEWQrrxVENGvZY2tqFUgZB9zbrpa4t4Rfo, ...";
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(event.target.value);
+  };
+
+  const validateAddresses = () => {
+    const delimitersRegex = /,|\s|\t|\n/; // Regex to match comma, space, tab, or newline
+    const addresses = inputText
+      .split(delimitersRegex)
+      .map((address) => address.trim());
+    const invalidAddressesList: string[] = [];
+    const validAddressesList: string[] = [];
+    let flag = 0;
+    addresses.forEach((address) => {
+      if (!address) {
+        // Skip empty addresses (remove empty strings from the list)
+        return;
+      }
+      if (
+        !isValidAddress(address, web3Account) ||
+        compareList.includes(address) ||
+        address === web3Address
+      ) {
+        flag = 1;
+        invalidAddressesList.push(address);
+      } else {
+        validAddressesList.push(address);
+      }
+    });
+    if (flag === 1) {
+      setIsExceptionAddressWrong?.(true);
```

Comment:
why the `?.` here? does this code behave as expected if `setIsExceptionAddressWrong` is not supplied? I'd prefer to make it a mandatory parameter that having an unusual optional call like this.

Reply:
Automatically it gets tested in Join call scenario